.github/workflows/ci.yml

```diff
@@ -333,33 +226,42 @@ jobs:
           pwd
           echo "$(pwd)\winfsp-test" >> "$GITHUB_PATH"
 
-      - uses: ./.github/actions/setup-rust
-        if: steps.python-changes.outputs.run == 'true'
-        with:
-          version: ${{ env.rust-version }}
-          cache-key: integration
-          os: ${{ matrix.os }}
-        timeout-minutes: 10
-
       - uses: ./.github/actions/setup-python-poetry
         if: steps.python-changes.outputs.run == 'true'
         id: setup-python
         with:
           python-version: ${{ env.python-version }}
           poetry-version: ${{ env.poetry-version }}
 
-      - name: Install python deps
-        if: steps.python-changes.outputs.run == 'true'
-        run: poetry install -E core -E backend
+      # libparsec is slow to compile, so we save it in cache and skip the
+      # compilation entirely if the Rust code hasn't changed !
+
+      - name: Use cached libparsec if Rust hasn't been modified
+        id: cache-libparsec
+        if: >-
+          steps.python-changes.outputs.run == 'true'
+          && steps.rust-changes.outputs.run != 'true'
+        uses: actions/cache@v3
+        with:
+          # Key cache contains a hash of all the files that are used to produce _parsec.so
+          # Hence if we have a cache hit we know that there is no need for a rebuild !
+          key: ${{ runner.os }}-${{ hashFiles('src/**', 'oxidation/libparsec/**', 'rust-toolchain.toml', 'Cargo.toml', 'Cargo.lock') }}-libparsec-python
```

Comment:
This also include platform dependent crate :thinking:
`platform_web` wouldn't be use by python :wink:  

Don't use `runner.os` that would break if we update the image used, use instead `matrix.os`

Reply:
done in 9d9bb67

Reply:
> This also include platform dependent crate 🤔
> platform_web wouldn't be use by python 😉

That's not a big deal for the moment. In the long run I think `platform_web`/`platform_native` will be removed given we will split the platform stuff per feature and not per platform (e.g. the `platform_storage` I was talking about the other day)

On top of that, I'd rather not play too clever on the regex: if we add a new crate we won't think about changing the regex, so we have to make sure things just work

> Don't use runner.os that would break if we update the image used, use instead matrix.os
👍

Reply:
> per feature

You mean per target ?

Reply:
My use of "feature" is misleading: I was referring to a component that need multiple implementation depending on the platform it is build for

For instance mountpoint on Linux/macOS sues fuse, on Windows WinFSP and Android API on Android. So we are going to put all of this in a `platform_mountpoint` crate that per-platform mostly separate codebases.

Reply:
Ok :+1: